sovrin_client/agent/endpoint.py

```diff
@@ -29,7 +29,8 @@ def __init__(self, port: int, msgHandler: Callable,
         if basedirpath:
             stackParams["basedirpath"] = basedirpath
 
-        super().__init__(stackParams, self.baseMsgHandler)
+        self.registry = {}
```

Comment:
I am not sure if we need to introduce this, as raet stack has a method findInRemotesByName which probably you can use to determine if a name is already used or not.